### ISSUE TITLE
How use the model.predict_generator to predict the probabilities of multi-labels

### ISSUE THREAD

**Issue by tianke0711** [keras-team/keras#9724]
Hi, there are 8 classes images, I want to use the predict_generator to predict the probabilities of multi-labels. How to get the probabilities of multi-labels of test images. Although I have get probabilities, I found that the sum of probabilities(such as 1.2) is more than 1 or less than 1. There may be wrong part in my code. Could you guys give your hand to help me.

```
[ 0.10360018  0.06169706  0.11487333  0.07720207  0.44261318  0.19750018
   0.08600973  0.22325319]
 [ 0.06560849  0.06978489  0.21796258  0.05673849  0.16526021  0.21255994
   0.09830733  0.24837734]
 [ 0.19014663  0.13065062  0.1236735   0.10173705  0.46399561  0.10760699
   0.05958129  0.12666202]]
..........................
```

```
def inceptionv3(img_dim, num_label):
    input_tensor = Input(shape=img_dim)
    base_model = InceptionV3(include_top=False,
                   weights='imagenet',
                   input_shape=img_dim)
    bn = BatchNormalization()(input_tensor)
    x = base_model(bn)
    x = GlobalAveragePooling2D()(x)
    x = Dropout(0.5)(x)
    output = Dense(num_label, activation='sigmoid')(x)
    model = Model(input_tensor, output)
    return model

```

```
def train_model(model, batch_size, epochs, img_size, x, y, test, n_fold, kf):
    roc_auc = metrics.roc_auc_score
    preds_train = np.zeros(len(x), dtype = np.float)
    preds_test = np.zeros((len(test),y.shape[1]), dtype = np.float)
    train_scores = []; valid_scores = []
    i = 1
    print('i is:', i)
    x = np.array(x)

       
        callbacks = [EarlyStopping(monitor='val_loss', patience=3, verbose=1, min_delta=1e-4),
             ReduceLROnPlateau(monitor='val_loss', factor=0.1, patience=1, cooldown=1, 
                               verbose=1, min_lr=1e-7),
             ModelCheckpoint(filepath='inception.fold_' + str(i) + '.hdf5', verbose=1,
                             save_best_only=True, save_weights_only=True, mode='auto')]

        train_steps = len(x_train) / batch_size
        valid_steps = len(x_valid) / batch_size
        test_steps = len(test) / batch_size
        model = model
        model.compile(optimizer=Adam(lr=1e-4), loss='categorical_crossentropy', 
                      metrics = ['accuracy'])

        model.fit_generator(train_generator(x_train, y_train, batch_size, img_size), train_steps, epochs=epochs, verbose=1, 
                            callbacks=callbacks, validation_data=valid_generator(x_valid, y_valid, batch_size, img_size), 
                            validation_steps=valid_steps)
        print('Running test predictions with fold {}'.format(i))

        preds_test_fold = model.predict_generator(generator=test_generator(test, batch_size, img_size),
                                              steps=test_steps, verbose=1)
        print('preds_test', preds_test.shape, preds_test_fold.shape, preds_test_fold)
        print('sum:', sum(preds_test_fold[1]))
        
        return preds_test_fold


```

`test_pred = train_model(model, batch_size, epochs, img_size, tr_p[0:50], tr_y[0:50], te[0:10], n_fold, kf)`

**Comment by Vijayabhaskar96** [comment]
You're using sigmoid as the activation function for your output layer, change it to softmax and the sum of the predictions will sum up to 1.

**Comment by tianke0711** [comment]
@Vijayabhaskar96  thanks for your kind response!  you are right, I have done it. By the way, could I ask you another question, I found that sometimes used the sigmoid and sometimes used the softmax in others code. Could you tell me when to use the softmax or sigmoid . Now I want deeply summary and share it.

**Comment by Vijayabhaskar96** [comment]
The final layer of your network(the output layer) should have sigmoid activation if you're task is to do binary classification (where 0 denotes a class and 1 denotes another class), if you are doing multiple classification then use softmax. But if you're solving a problem where multiple output is possible then use sigmoid for the output layer,so that it can predict multiple categories at the same time(like you did,so that the probabilities don't sum upto 1, because each number in the output list is a probability of its own).
